File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.c
bool searchMatrix(int** matrix, int matrixSize, int* matrixColSize, int target) {
    int row = matrixSize - 1, col = 0;

    while (row >= 0 && col < *matrixColSize) {
        if (matrix[row][col] == target) {
            return true;
        }
        else if (matrix[row][col] > target) {
            row--;
        }
        else {
            col++;
        }
    }

    return false;
}

File: 0240-search-a-2d-matrix-ii/0240-search-a-2d-matrix-ii.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int trow=matrix.length;
        int tcol=matrix[0].length;
        int row=0,col=tcol-1;
        while(row<trow && col>=0){
            if(matrix[row][col]==target){
                return true;
            }else if(matrix[row][col]>target){
                col--;
            }else{
                row++;
            }
        }return false;
    }
}